### DISCUSSION TITLE
Evidence Monorepo Support

### DISCUSSION THREAD

**Discussion by archiewood** [evidence-dev/evidence#2375] [General]
Ability to have multiple evidence projects that live inside one repo and have the dev experience run smoothly.

- components in one place only but available across projects
- can run commands from the root and have them do the right things for multiple projects
- (maybe) some recommendation on how to kick off builds in cases where you have >1

**Comment by AlbinXXX** [comment]
do you mind if I work on the following issue:
I want to be able to run dev environment from the monorepo root folder and be able to run it all with one or two commands?

**Comment by ItsMeBrianD** [comment]
I think this issue is related to working on evidence projects (not the evidence monorepo itself) but if you want to take a stab at improving the dev experience in this repository we're always open to pull requests!

**Comment by shakedlokits** [comment]
This is what blocks us from using evidence.
We have an Nx monorepo of our project and you can't run evidence in it. It breaks the CLI.

**Comment by archiewood** [comment]
What's your setup?

Can you `npm --prefix ./my-evidence-folder run sources` etc?

**Comment by shakedlokits** [comment]
Hey @archiewood 
It's an Nx monorepo. It has a directory called `sites` and I created a `monorepo/sites/business-analysis` directory for evidence.

This is what I get:
```bash
➜  npm --prefix ./sites/business-analysis run sources

> my-evidence-project@0.0.1 sources
> evidence sources

Could not find evidence plugins file. (Look at /Users/shakedlokits/Documents/Projects/monorepo/evidence.plugins.yaml)
```

It's looking in the wrong directory for the `evidence.plugins.yaml` file, even though I gave it a prefix.

**Comment by archiewood** [comment]
Which folder did you store your evidence.plugins.yaml in? Root or business-analysis?

**Comment by shakedlokits** [comment]
I stored it in `monorepo/sites/business-analysis` as it is unrelated to the other teams in the company.

**Comment by shakedlokits** [comment]
@archiewood any updates?

**Comment by archiewood** [comment]
Don't have a fix slated yet

in the meantime are you able to `cd` into the correct directory and run from there?

**Comment by archiewood** [comment]
Will also happily review PRs if this feature is important to you!

**Comment by hughess** [comment]
This may work if you use the VS Code extension's built-in commands. If you are in your monorepo root and click Start Evidence (or use `Cmd/Ctrl + R`) it should start evidence in the subfolder it's in

**Comment by alex-laycalvert** [comment]
I'm having an issue that I believe is related to monorepo support but not exactly the same as what's been discussed here. I used `npx degit` to create the evidence app within my turborepo project (at `<my_project>/apps/frontend`), navigating to `apps/frontend` and running `npm run sources` works fine with no errors, but when running with `npm run dev`, the web server just results in a 404 when going to `localhost:3000`. It's definitely running on that port after investigating, but can't find any pages.

Installing/running evidence in a standalone repo works fine, maybe has something to do with where the dependencies of the monorepo are being hoisted?

(I've also attempted running at the root of my monorepo with `--prefix` with no success)
